testing/microlith-test/integration/prometheus_alert_overrides/prometheus_alert_overrides.bats

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env bats
+
+# Copyright 2021 Couchbase, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file  except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the  License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+load "$BATS_SUPPORT_ROOT/load.bash"
+load "$BATS_ASSERT_ROOT/load.bash"
+load "$BATS_FILE_ROOT/load.bash"
+load "../helpers"
+
+export CMOS_IMAGE=${CMOS_IMAGE:-couchbase/observability-stack:v1}
+
+setup() {
+    if [ "$TEST_INTEGRATION" == "true" ]; then
+        skip "Skipping integration tests"
+    fi
+    find couchbase custom overrides -regex '.*\.ya?ml\.orig$' -delete
+}
+
+teardown() {
+    if [ "$SKIP_TEARDOWN" == "true" ]; then
+        echo "# Skipping teardown. Make sure to manually run the commands in teardown()." >&3
+        return
+    fi
+    docker-compose rm -v --force --stop
+}
+
+@test "Alert overrides (generated YAML file)" {
+    docker-compose up -d --force-recreate --remove-orphans
+    run docker-compose exec cmos cat /etc/prometheus/alerting/generated/alerts.yaml
+    assert_line -p 'expr: untouched'
+    assert_line -p 'expr: overridden{foo!="true"}'
+    assert_line -p 'expr: disabled{foo!="true"}'
+    refute_line -p 'expr: disabled{}'
+    assert_line -p 'expr: overridden{foo="true"}'
+}
+
+@test "Alert overrides (API JSON output)" {
+    docker-compose up -d --force-recreate --remove-orphans
+    sleep 5 # ensure Prom has a chance to start up
```

Comment:
I think we should have a helper method for this, poll a number of times with a delay between each until it is ready or fail.

```
    URL=$1
    MAX_ATTEMPTS=$2
    ATTEMPTS=0
    until curl -s -o /dev/null "${URL}"; do
        if [ $ATTEMPTS -gt $MAX_ATTEMPTS ]; then
            assert_failure "unable to communicate with $URL"
        fi
        ((ATTEMPTS++))
        sleep 10
    done
```